From 3f80764170edee7e3eba875a0551ee2a5de85d2a Mon Sep 17 00:00:00 2001
From: madoka773 <valigarmanda55@gmail.com>
Date: Tue, 26 Dec 2023 23:46:20 +0800
Subject: [PATCH] fix electron exec

---
 packages/desktop/main/db.ts | 21 +--------------------
 1 file changed, 1 insertion(+), 20 deletions(-)

diff --git a/packages/desktop/main/db.ts b/packages/desktop/main/db.ts
index 931c5fa..90286c3 100644
--- a/packages/desktop/main/db.ts
+++ b/packages/desktop/main/db.ts
@@ -88,10 +88,7 @@ class DB {
 
     try {
       createFileIfNotExist(this.dbFilePath)
-
-      this.sqlite = new SQLite3(this.dbFilePath, {
-        nativeBinding: this.getBinPath(),
-      })
+      this.sqlite = new SQLite3(this.dbFilePath)
       this.sqlite.pragma('auto_vacuum = FULL')
       this.initTables()
       this.migrate()
@@ -103,22 +100,6 @@ class DB {
     }
   }
 
-  private getBinPath() {
-    console
-    const devBinPath = path.resolve(
-      app.getPath('userData'),
-      `../../bin/better_sqlite3_${os.platform}_${os.arch}.node`
-    )
-    const prodBinPaths = {
-      darwin: path.resolve(app.getPath('exe'), `../../Resources/bin/better_sqlite3.node`),
-      win32: path.resolve(app.getPath('exe'), `../resources/bin/better_sqlite3.node`),
-      linux: path.resolve(app.getPath('exe'), `../resources/bin/better_sqlite3.node`),
-    }
-    return isProd
-      ? prodBinPaths[os.platform as unknown as 'darwin' | 'win32' | 'linux']
-      : devBinPath
-  }
-
   initTables() {
     log.info('[db] Initializing database tables...')
     const init = readSqlFile('init.sql')
-- 
2.43.0

